lib/screens/v2/candidate_generator/dup/alerts.ex

```diff
@@ -104,6 +104,7 @@ defmodule Screens.V2.CandidateGenerator.Dup.Alerts do
         %DupAlert{
           screen: config,
           alert: alert,
+          alert_effect_detailed: detailed_alert_effect(alert, location_context),
```

Comment:
It seems like we could do this logic entirely inside the `DupAlert` widget, and dispense with the extra field passed into it. The only inputs are the `alert` (which is already a field passed in) and `location_context` (ditto). Then we also don't need to define this "API" between the candidate generator and the widget; the widget can do whatever is most convenient internally.